stix_shifter/stix_transmission/src/modules/guardium/guardiumapiclient.py

```diff
@@ -0,0 +1,448 @@
+from ..utils.RestApiClient import RestApiClient
+from ..utils.RestApiClient import ResponseWrapper
+from requests.models import Response
+import urllib.parse
+import logging
+import sys
+import json
+import datetime
+import os
+import base64
+
+
+class APIClient():
+
+    # API METHODS
+
+    # These methods are used to call Ariel's API methods through http requests.
+    # Each method makes use of the http methods below to perform the requests.
+
+    # This class will encode any data or query parameters which will then be
+    # sent to the call_api() method of the RestApiClient
+    def __init__(self, connection, configuration):
+#
+        self.endpoint_start = 'restAPI/'
+        self.connection = connection
+        self.configuration = configuration
+        self.headers = dict()
+        self.search_id = None
+        self.query = None
+        self.authorization = None
+        self.credential = {}
+
+#
+# Check if connection object contains the following
+ 
+        username = connection.get("username", None)
+        password = connection.get("password", None)
+        grant_type = connection.get("grant_type", None)
+        client_id = connection.get("client_id", None)
+        client_secret = connection.get("client_secret", None)
+        access_token = connection.get("guardium_access_token", None)
+        expiresTimestamp = connection.get(
+            "guardium_token_expiryTimestamp", None)
+#
+#       if the connection object does not contain the credential -- it can be picked from this folder
+#       this may not be implemented.
+        self.credential_filepath = "./stix_shifter/stix_transmission/src/modules/guardium/credentials/"
```

Comment:
Credentials will always come from the configuration.

Reply:
Let me remove and throw error if the connector does not receive all 5 params for the credential.